tap_jira/streams.py

```diff
@@ -174,11 +173,8 @@ def sync(self):
                                         "/rest/api/2/group/member",
                                         params=params):
                     self.write_page(page)
-            except requests.exceptions.HTTPError as http_error:
-                if http_error.response.status_code == 404:
-                    LOGGER.info("Could not find group \"%s\", skipping", group)
-                else:
-                    raise http_error
+            except JiraNotFoundError:
+                LOGGER.info("Could not find group \"%s\", skipping", group)
```

Comment:
@dbshah1212 What if the exception is other than 404. Looks like the updated try-except block is not handling the scenario where status_code!= 404

Reply:
Also, I did not see any test case related to this code change

Reply:
@karanpanchal-crest , As we discussed, other exceptions will be raised directly, and the skipping log will be printed in the JiraNotFoundError case and I also added unit tests for the above change.